lib/modules/platform/codecommit/index.ts

```diff
@@ -0,0 +1,693 @@
+import { PullRequestStatusEnum } from '@aws-sdk/client-codecommit';
+import type { Credentials } from '@aws-sdk/types';
+import JSON5 from 'json5';
+import {
+  PLATFORM_BAD_CREDENTIALS,
+  REPOSITORY_EMPTY,
+  REPOSITORY_NOT_FOUND,
+} from '../../../constants/error-messages';
+import { logger } from '../../../logger';
+import { BranchStatus, PrState, VulnerabilityAlert } from '../../../types';
+import * as git from '../../../util/git';
+import { regEx } from '../../../util/regex';
+import { sanitize } from '../../../util/sanitize';
+import type {
+  BranchStatusConfig,
+  CreatePRConfig,
+  EnsureCommentConfig,
+  EnsureCommentRemovalConfig,
+  EnsureIssueConfig,
+  EnsureIssueResult,
+  FindPRConfig,
+  Issue,
+  MergePRConfig,
+  PlatformParams,
+  PlatformResult,
+  Pr,
+  RepoParams,
+  RepoResult,
+  UpdatePrConfig,
+} from '../types';
+import { smartTruncate } from '../utils/pr-body';
+import * as client from './codecommit-client';
+import { getCodeCommitUrl, getNewBranchName } from './util';
+
+const decoder = new TextDecoder();
+
+interface Config {
+  repository?: string;
+  defaultBranch?: string;
+  region?: string;
+  prList?: Pr[];
+  credentials?: Credentials;
+}
+
+const config: Config = {};
+
+export function initPlatform({
+  endpoint,
+  username,
+  password,
+}: PlatformParams): Promise<PlatformResult> {
+  let accessKeyId = username;
+  let secretAccessKey = password;
+  let region;
+
+  if (!accessKeyId) {
+    accessKeyId = process.env.AWS_ACCESS_KEY_ID;
+  }
+  if (!secretAccessKey) {
+    secretAccessKey = process.env.AWS_SECRET_ACCESS_KEY;
+  }
+  if (endpoint) {
+    const regionReg = regEx(/.*codecommit\.(?<region>.+)\.amazonaws\.com/);
+    const codeCommitMatch = regionReg.exec(endpoint);
+    region = codeCommitMatch?.groups?.region;
+    if (!region) {
+      logger.warn("Can't parse region, make sure your endpoint is correct");
+    }
+  } else {
+    region = process.env.AWS_REGION;
+  }
+
+  if (!accessKeyId || !secretAccessKey || !region) {
+    throw new Error(
+      'Init: You must configure a AWS user(accessKeyId), password(secretAccessKey) and endpoint/AWS_REGION'
+    );
+  }
+  config.region = region;
+  const credentials = {
+    accessKeyId: accessKeyId,
+    secretAccessKey: secretAccessKey,
+  };
+  config.credentials = credentials;
+
+  client.buildCodeCommitClient(region, credentials);
```

Comment:
Does this test whether the credentials *work*? We should test if they work within this initPlatform. And if so there needs to be handling for that, such as error catching if it throws

Reply:
at first i thought there was no [API ](https://docs.aws.amazon.com/codecommit/latest/APIReference/Welcome.html) in codecommit that can work without the repository input, so i verified in `InitRepo()`.
 `git.clone` inside ,
 now that i think of it, the only API that i can call and it doesn't need the repo input is : `listRepositories`, 
 should i call that and handle? or can we handle on listRepos(auto discover) or initRepo(normal run)

Reply:
Ideally we want to call *something* during initPlatform which lets us verify that the credentials are valid

Reply:
Did u know that aws start charging after 10k calls per month ? Maybe we should consider that 
But u might be right because after i tested that i found out that git.clone is retryinf to connect 5 times.. 
So if creds are bad we have 5 calls in this case but on actual runs we dont pay that extra check call.. xD not sure whats better

Reply:
We should not be worried about a single API call in initPlatform(), which is called once per Renovate run. A single run with multiple repos will end up using hundreds or thousands in most cases. The benefit to the user of finding the platform credentials error immediately is worth it